src/App.tsx

```diff
@@ -1,26 +1,55 @@
+import { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [isLoading, setIsLoading] = useState(true);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        dispatch(todosSlice.actions.setTodos(todosFromServer));
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchTodos();
+  }, [dispatch]);
+
+  return (
+    <>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="section">
+          <div className="container">
+            <div className="box">
+              <h1 className="title">Todos:</h1>
+
+              <div className="block">
+                <TodoFilter />
+              </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+              <div className="block">
+                {todos.length < 0 && <Loader />}
```

Comment:
The condition `todos.length < 0` will never be true because the length of an array cannot be negative. If you want to show the `Loader` when there are no todos, you should use `todos.length === 0`.